src/main/kotlin/org/rust/cargo/toolchain/tools/Cargo.kt

```diff
@@ -104,7 +104,7 @@ class Cargo(
         companion object {
             fun from(line: String): BinaryCrate {
                 val name = line.substringBefore(' ')
-                val rawVersion = line.substringAfter(' ').removePrefix("v").removeSuffix(":")
+                val rawVersion = line.substringAfter(' ').removePrefix("v").removeSuffix(":").substringBefore('(').trim()
```

Comment:
Actually, it becomes too complicated. Maybe provide a regex to handle proper lines?

Something like

```kotlin
// Examples:
// cargo-expand v1.0.16:
// wasm-pack v0.10.3 (/home/josh/dev/wasm-pack):
private val VERSION_LINE: Regex = """(?<name>[\w-]+) v(?<version>\d+\.\d+\.\d+).*:""".toRegex()

fun from(line: String): BinaryCrate? {
    val result = VERSION_LINE.matchEntire(line) ?: return null
    val name = result.groups["name"]?.value ?: return null
    val rawVersion = result.groups["version"]?.value ?: return null
    return BinaryCrate(name, SemVer.parseFromText(rawVersion))
}
```

in this case, it's clearer what we expect from a line

Reply:
Awesome suggestion, I'll take a look at it. I was also curious when the line would end up being too complicated.

Yeah, it would be really nice if there was a parsable format for this, as you always end up in this situation :facepalm:.

Reply:
It seems like the regex is working, so I applied the suggestion and rebased the PR.